mesonbuild/build.py

```diff
@@ -112,7 +112,7 @@ class DFeatures(TypedDict):
     cs_kwargs)
 
 known_exe_kwargs = known_build_target_kwargs | {'implib', 'export_dynamic', 'pie', 'vs_module_defs'}
-known_shlib_kwargs = known_build_target_kwargs | {'version', 'soversion', 'vs_module_defs', 'darwin_versions', 'rust_abi'}
+known_shlib_kwargs = known_build_target_kwargs | {'version', 'soversion', 'vs_module_defs', 'darwin_versions', 'rust_abi', 'shortname'}
```

Comment:
'shortname' needs to be added to the documentation for shared_library in the docs section.

Additionally, in interpreter/type_checking.py you need to add:
```python
KwargInfo('shortname', (str, NoneType), since='1.8')
```
to the `_EXCLUSIVE_SHARED_LIB_KWS` list.

Reply:
Ok. I'll try.

Reply:
I think I did this at https://github.com/mesonbuild/meson/pull/14106/commits/d33570793432578233d1b07005788abc2394d9f9#diff-b527ff916e48b4d5638516b78dcff9b4650d1ccd7271e00f5c52bc851e2cc578L786-L791 and https://github.com/mesonbuild/meson/pull/14106/commits/d33570793432578233d1b07005788abc2394d9f9#diff-61272a50bc50fa76acf2d43681320dc545098a2e1a769cd747e23f1e063c2a67L54-R62.

There are additional things?